[ADD] *: implement and Enhance Various Features for Practice

rpka-odoo opened the pull request:
PR Summary

This PR introduces multiple practice tasks related to Odoo development, covering different modules and functionalities.
Changes

event_ticket (Task ID: 4598797)

    Adds tickets_per_registration field in event.event.ticket.
    Limits ticket selection based on event max, ticket type max, and registration limit.

product_warranty (Task ID: 4602871)

    Implements a product warranty management module with a wizard.
    Adds warranty selection in sales and ensures security & test coverage.

product_catalog_redesign (Task ID: 4613280)

    Enhances product image view with larger preview on click.
    Ensures responsiveness and prevents unintended quantity additions.

rpka-odoo committed:
[ADD] event_ticket: limit tickets per registration

PURPOSE:
Restrict the number of tickets a user can register per event,
ensuring better event capacity management and preventing overbooking.

SPECIFICATION:
- Introduce a field to define ticket limits per registration.
- Display ticket limits in the event ticket list view.
- Adjust the ticket selection modal to enforce limits.
- Apply constraints based on event capacity, ticket type, and defined limits.
- Allow up to 9 tickets if no specific limit is set.

Task-4598797

rpka-odoo committed:
[ADD] product_warranty: implement product warranty module with wizard

PURPOSE:
Introduce a product warranty management system, allowing users to configure
warranties and add them to sale orders through a guided wizard.

SPECIFICATION:
- Allow users to define and manage product warranties.
- Enable selecting and adding warranties to sale orders via a wizard.
- Automatically fetch related products when opening the wizard.
- Ensure warranty addition as a sale order line upon confirmation.
- Restrict access with proper security rules.
- Provide a dedicated menu under Sales > Configuration.
- Include test cases for wizard functionality and warranty addition.
- Add demo data with sample warranties and sale orders.

Task-4602871

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/tutorials/pull/415.png)](https://mergebot.odoo.com/odoo/tutorials/pull/415)

aktr-odoo commented:
Give proper commit message

rpka-odoo commented:
@aktr-odoo Requested changes have been made can you please have a look?

aktr-odoo commented on product_warranty/models/sale_order_line.py:
```diff
@@ -0,0 +1,8 @@
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    linked_to_warranty = fields.One2many("sale.order.line", "warranty_linked_with")
+    warranty_linked_with = fields.Many2one("sale.order.line", ondelete="cascade")
```
We already have this relation available in the model.
Any reason to define it again?

rpka-odoo replied:
Yes sir, first I had no idea about one, so I defined it. Now I know about it and have removed it.

aktr-odoo reviewed (changes requested):
:wave: 

I have left a few comments. Could you please have a look?

rpka-odoo committed:
[ADD] product_catalog_redesign: enhance product image view in catalog

PURPOSE:
Enhance the product catalog by allowing users to view larger product images,
improving accessibility and usability, especially on mobile.

SPECIFICATION:
- Clicking a product image opens a larger view.
- On mobile, images open in an expanded view for better visibility.
- Prevent accidental quantity addition when clicking images.
- Ensure a smooth and responsive experience on all devices.
- Add a close button for easy dismissal of enlarged images.

Task-4613280

rpka-odoo commented:
@aktr-odoo Requested changes have been made can you please have a look?

aktr-odoo commented on product_warranty/views/sale_menus.xml:
Give proper file name

aktr-odoo commented on product_warranty/tests/test_product_warranty.py:
All the best cases are included in one file.
The test cases must be in a separate file, code-wise.

aktr-odoo commented on product_warranty/wizard/add_warranty.py:
Give proper file name

aktr-odoo commented on product_warranty/wizard/add_warranty_line.py:
Give proper file name

aktr-odoo commented on product_warranty/wizard/add_warranty_wizard.xml:
Give proper file name

aktr-odoo commented on product_warranty/wizard/add_warranty.py:
```diff
@@ -0,0 +1,67 @@
+        if not order_id:
+            return res
+
+        order = self.env["sale.order"].browse(order_id)
+
+        if not order or not order.exists():
+            return res
```
Do we need `if not order`? 

Also it seems like `.exists()` can be used properly

rpka-odoo replied:
Yes I understand I change it , 

if not order? is not need here and exists() method is check whether the record(s) in the recordset actually exist in the database

aktr-odoo commented on product_warranty/wizard/add_warranty.py:
```diff
@@ -0,0 +1,67 @@
+
+        warranty_lines = []
+        for line in order.order_line:
+            if line.product_id.is_warranty and not line.linked_line_ids:
```
Do we have any case where there lines will have `linked_line_ids` up to this part?

aktr-odoo reviewed (changes requested):
:wave: 

I have left a few comments. Could you please have a look?

rpka-odoo commented:
@aktr-odoo Requested changes have been made can you please have a look?

rpka-odoo committed:
[ADD] pos_receipt: configure receipt layout

Purpose:
Now users can configure their receipt format for each POS configuration.

Specification:
The following options are editable with this wizard:
- Receipt Layout:
  Users can choose the overall look of the receipt.
  Available options are: Light, Boxed, Lined
- Receipt Logo: A logo displayed at the top of the receipt.
- Header and Footer: Personalize the receipt with custom header & footer content
